### PR TITLE
Http log setting

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

**Comment by TheOneRing** [comment]
https://github.com/owncloud/client/issues/8073

File: src/libsync/configfile.cpp [hunk 1]
```diff
@@ -113,6 +116,13 @@ ConfigFile::ConfigFile()
 
     QSettings settings(config, QSettings::IniFormat);
     settings.beginGroup(defaultConnection());
+
+    // run init only once
+    static bool init = [this]() {
+        setLogHttp(logHttp());
+        return false;
+    }();
+    Q_UNUSED(init);
```
Comment (dschmidt) [inline src/libsync/configfile.cpp:125]: I really dislike this singleton-ish behavior. 

Can't you call this explicitly from main.cpp or from application or something?

File: src/libsync/logger.cpp [hunk 2]
```diff
@@ -210,14 +201,18 @@ void Logger::setLogFlush(bool flush)
...
+        removeLogRule(rules);
+    }
     _logDebug = debug;
 }
 
 QString Logger::temporaryFolderLogDirPath() const
 {
-    QString dirName = APPLICATION_SHORTNAME + QString("-logdir");
-    return QDir::temp().filePath(dirName);
+    return QDir::temp().filePath(QStringLiteral(APPLICATION_SHORTNAME "-logdir"));
```
Comment (dschmidt) [inline src/libsync/logger.cpp:215]: :+1:

File: src/libsync/configfile.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/libsync/configfile.cpp:125]: Well yes with 2.8 I will rework the whole class.
For now I'd like to just do this...

**Pull request merged**